admin/class-acc_user_importer-admin.php

```diff
@@ -277,77 +277,22 @@ private function proccess_user_data ( $users ) {
 			return $api_response;
 		}
 		
-		//remove record entirely if membership number is missing
-		$malformed_users = [];
-		foreach ( $users as $key => $user ) {
-			$member_id = $user['MEMBERSHIP_N'];
-			if (! (is_numeric($member_id) )) {
-				$malformed_users[] = $user;
-				unset( $users[$key] );
-				continue;
-			}
-		}
-		
-		//log outcome of removing users without a membership number
-		if (count($malformed_users) > 0) {
-			$this->log_dual("Removed " . count($malformed_users) . " record(s) without membership numbers.");
-			foreach ( $malformed_users as $id => $user ) {
-				$this->log_dual("  [" . ($id + 1) . "] " . $user['FirstName'] . " " . $user['LastName']);
-			}
-		}
-		
-		//sanitize - i.e. remove keys/values that aren't in this list
-		foreach ( $users as $key => $user ) {
-			
-			$allowed_columns = array(	//'PRODUCT_CODE',
-										//'MemberType',
-										'Membership Expiry Date',
-										'Email',
-										'Cell Phone',
-										'HomePhone',
-										//'Address1',
-										//'Address2',
-										'City',
-										'LastName',
-										'FirstName',
-										//'Contact ID',
-										'MEMBERSHIP_N'
-									);
-									
-			foreach ( $user as $value_key => $value ) {
-				
-				if (! in_array( $value_key, $allowed_columns ) ) {
-					unset( $user[$value_key] );
-				}
-				$users[$key] = $user; //save into correct scope
-			}
-		}
```

Comment:
From what I understand, the code removes keys we do not care about like 'Address1'.  Why spend cycles removing something we will ignore anyway?  Is this why the code is being removed?  If so, great!

Reply:
Correct, I could not find a reason why we need to remove those keys if we build the array used for updating the member below.